### PR TITLE
Make the experience better when start-pspester doesn't find pester

### PR DESCRIPTION
**Pull request by TravisEz13**
## PR Summary

- refactor code to restore pester into a separate function called Restore-PSPester
- update message on what to do when pester is missing
- Add ability for `get-psoptions` to default to `new-psoptions`
- fix an issue with `publish-pstesttools` when a build has not been run since build.psm1 has been imported (try to use the default options)
- make `start-pspester` use the last build, not just use the default options
- fix an issue in restore caused some files not to be removed 
## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [NA] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [NA] Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [NA] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [NA] [Add `[feature]` if the change is significant or affectes feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

**Review by markekraus — APPROVED** [review]
LGTM and Thank you!!!

**Comment by SteveL-MSFT** [comment]
Can you address https://github.com/PowerShell/PowerShell/issues/5640 in the same PR?

**Comment by TravisEz13** [comment]
Added two commits for the docs.  One to make them pass unit tests and one to make the updates for easier review.

**Review by markekraus — APPROVED** [review]

File: build.psm1 [hunk 1]
```diff
@@ -963,8 +984,8 @@ function Start-PSPester {
     {
         Write-Warning @"
 Pester module not found.
-Make sure that the proper git submodules are installed by running:
-    git submodule update --init
+Restore to module to '$Pester' by running:
```
Comment (SteveL-MSFT) [inline build.psm1]: Is the first `to` extra?

**Review by SteveL-MSFT — CHANGES_REQUESTED** [review]

**Review by SteveL-MSFT — APPROVED** [review]

**Pull request merged**